### PR TITLE
Set up continuous deployment using encrypted secret_files

### PR DESCRIPTION
**Pull request by oshoham**
Set up continuous deployment using Travis CI.

**Benefits:**
- No longer need to run deployment scripts locally to deploy to staging and production

**Downsides:**
- Need to check encrypted tar of secret_files into the repo, and make sure to update it when contents change.

The need to check an encrypted tar of secret_files into the repo didn't occur to me until I was close to done implementing this, and I'm not wild about it, b/c it negates some of the convenience of the whole continuous deployment thing.

@mhfowler @jayres I'm down to scrap this if you don't think it's worth it, but the work is done so I figured I might as well put it in front of you.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,3 +30,23 @@ script:
...
+before_deploy:
+  - sudo apt-get install python-dev -y
+  - cd $TRAVIS_BUILD_DIR/backend && pip install -r requirements.txt
+  - cd $TRAVIS_BUILD_DIR/frontend && npm install && cd $TRAVIS_BUILD_DIR
+  - openssl aes-256-cbc -K $encrypted_5d10786c9aae_key -iv $encrypted_5d10786c9aae_iv -in backend/devops/secret_files.tar.enc -out backend/devops/secret_files.tar -d
+  - cd $TRAVIS_BUILD_DIR/backend/devops && tar xvf secret_files.tar && cd $TRAVIS_BUILD_DIR
+
+deploy:
+  - provider: script
+    script: backend/bash/staging_quick_deploy.sh
```
Comment (mhfowler) [inline .travis.yml]: staging_quick_deploy also passes a tag --quick
which runs a subset of the ansible commands, which probably isn't what we want here

guessing that you made this quick_deploy to make it avoid running the tests?

If so my recommendation is to remove the tests from staging_deploy.sh and then have this call staging_deploy.sh

**Comment by mhfowler** [comment]
this is very cool

will travis also send slack messages when it is deploying? that would be helpful

File: .travis.yml (hunk 1 above)
Comment (oshoham) [inline .travis.yml]: good call, I'll do that

**Comment by oshoham** [comment]
@mhfowler I can configure Travis to log to Slack, I'll add that to this PR

**Pull request merged**